### DISCUSSION TITLE
Could ~/.config/harlequin/config.toml also be an option on Mac?

### DISCUSSION THREAD

**Discussion by runeflobakk** [tconbeer/harlequin#633] [Q&A, Troubleshooting, Support] [answered]
[Per the documentation](https://harlequin.sh/docs/config-file), on Macs, the config file is tried resolved from `~/Library/Application Support/harlequin/config.toml`. Would it be possible to include `~/.config/harlequin/config.toml` as a possible location which will be searched as the "user’s default config directory" on Macs as well?

I already have config for various tools under `~/.config`, and it is nice to have them in this folder to avoid cluttering my home dir with numerous config files, even if they are dot-files and hidden by default. Using `~/.harlequin.toml` works as it should though.

I suspect the `~/Library/Application Support/` are rarely considered as a convenient config location for _terminal_ apps, even by Mac users, though I can of course only speak for myself.

**Comment by tconbeer** [comment, marked as answer]
We follow platformdirs here, and while this is a little [contentious](https://github.com/platformdirs/platformdirs/issues/4), the Mac Standard is `~/Library/Application Support`.

You can invoke Harlequin with the `--config-path` option to specify a different config path. Maybe we should add a `HARLEQUIN_CONFIG_PATH` environment variable that would allow you to set that once? I'd be open to that (it's a one-line PR to add `envvar="HARLEQUIN_CONFIG_PATH"` here):

https://github.com/tconbeer/harlequin/blob/a9a525d6e5411d7a0d85ab61ccf8ed66ce632bfb/src/harlequin/cli.py#L172-L173

**Reply by runeflobakk** [reply]
That's fair. I really do not wish to go into that discussion of principles. (didn't notice this existing issue on the matter)
I'll settle for keeping `.harlequin.toml` in my home dir.
